geoviews/plotting/__init__.py

```diff
@@ -145,7 +146,18 @@ class GeoImagePlot(GeoPlot, ColorbarPlot):
     def get_data(self, element, ranges, style):
         self._norm_kwargs(element, ranges, style, element.vdims[0])
         style.pop('interpolation')
-        return (element.data.copy(),), style, {}
+        cube = element.data.copy()
+        # Make sure both coordinates have bounds to avoid iris warning.
+        for coord in cube.dim_coords:
+            if not coord.has_bounds():
+                coord.guess_bounds()
+                # Clip auto-generated latitude bounds to +/-90 to avoid
+                # cartopy warning.
+                if 'latitude' in coord.name():
+                    points = coord.points
+                    if (points >= -90).all() and (points <= 90).all():
+                        coord.bounds = np.clip(coord.bounds, -90, 90)
```

Comment:
The Iris PR has been merged and it'll turn up in the next release, so I need to delete the code from here.